### DISCUSSION TITLE
which object i should use for wireless button

### DISCUSSION THREAD

**Discussion by chenliuyang1989** [bacnet-stack/bacnet-stack#373] [Q&A] [answered]
Hi，now i have a wireless button to support bacnet，it has click ,double click  and long press click event.
and i  am confused to select which object,
MI is not appropriate i think，do you have some advices，thanks

**Comment by stargieg** [comment, marked as answer]
I would take 3 BI (one for each event). BI are easier to process further.
MI are for defined system states e.g. Forward,Reverse,Stop
Greetings and A successful new year

**Comment by chenliuyang1989** [comment]
Thanks for your quick response，and Happy new  year！
This is one better option than MI。

These states are edge triggered which means the status will not keeping。
This is just status at one moment，and can also call just one event, is there any other options?
Thanks

**Comment by stargieg** [comment]
You can optionally add a counter so you can detect missing events, reboots or network outages.
my2cents

**Comment by chenliuyang1989** [comment]
ok。Thanks

**Comment by skarg** [comment]
A BACnet Multi-state Input object could also be used to help reduce network jitter which can occur with only momentary button press-release states that rely on timing between press/release.  States could be:
- released = 1
- press = 2
- double press = 3
- hold = 4

If the button is not momentary and is instead a toggle button, the BACnet Binary Input can be reliably used and press/release simply toggles the state active/inactive for every release.

**Comment by chenliuyang1989** [comment]
Hi, it's been a long time since we last got in touch
There are some other questions
now i want transform matter(https://github.com/project-chip/connectedhomeip) device to Bacnet.
Matter have event (eg: for motion sensor ), which object is the right i should to choose to represent event(motion sensor ,) ? Thanks
